### PR TITLE
DROOLS-3590/DROOLS-3589: [DMN Designer] Data Types - When users create a Data Type, they must be able to perform shortcut operations in the last created DT /  Shortcut tooltip remains opened in some scenarios

### PR DESCRIPTION
**Pull request by karreiro**
See:
- [DROOLS-3590](https://issues.jboss.org/browse/DROOLS-3590): [DMN Designer] Data Types - When users create a Data Type, they must be able to perform shortcut operations in the last created DT
- [DROOLS-3589](https://issues.jboss.org/browse/DROOLS-3589): [DMN Designer] Data Types - Shortcut tooltip remains opened in some scenarios

---

**DROOLS-3590**

---

**DROOLS-3589**

Before:

After:

---

Part of an ensemble:
- https://github.com/kiegroup/appformer/pull/630
- https://github.com/kiegroup/kie-wb-common/pull/2466

### CODE REVIEW

**Comment by karreiro** [comment]
Jenkins please retest this.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeList.java [hunk 1]
```diff
@@ -48,6 +49,8 @@
 
     private final DataTypeStackHash dataTypeStackHash;
 
+    private List<Consumer<DataTypeListItem>> dataTypeListItemUpdateCallbacks = new ArrayList<>();
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeList.java]: This class is `@ApplicationScoped`  and this collection keeps growing and growing the more times I create a DMN file.

**Comment by jomarko** [comment]
Just for record, button hints don't stay rendered anymore, this part of fix works perfect.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeList.java (hunk 1 above)
Comment (karreiro) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeList.java]: Thank you, @manstis. I was doing some premature optimization here. I've taken a step back, and removed the array to fix it ;-)

**Comment by karreiro** [comment]
Jenkins please retest this.

**Comment by karreiro** [comment]
Jenkins please retest this.

**Comment by karreiro** [comment]
Jenkins please retest this.

**Comment by karreiro** [comment]
Jenkins please retest this.

**Pull request merged**